File: e2e/java/src/test/java/dev/kreuzberg/e2e/RateLimitTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: rate_limit. */
class RateLimitTest {
    @Test
    void testRateLimitBasicDelay() throws Exception {
        // Rate limiter adds delay between requests to the same domain
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'crawl.pages_crawled' not available on result type
        // skipped: field 'rate_limit.min_duration_ms' not available on result type
    }

    @Test
    void testRateLimitZeroNoDelay() throws Exception {
        // Rate limiter with zero delay does not slow crawling
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'crawl.pages_crawled' not available on result type
    }

}

File: e2e/go/rate_limit_test.go
// E2e tests for category: rate_limit
package e2e_test

import (
	"strings"
	"testing"

	pkg "github.com/kreuzberg-dev/kreuzcrawl"
)

func Test_RateLimitBasicDelay(t *testing.T) {
	// Rate limiter adds delay between requests to the same domain
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'crawl.pages_crawled' not available on result type
	// skipped: field 'rate_limit.min_duration_ms' not available on result type
}

func Test_RateLimitZeroNoDelay(t *testing.T) {
	// Rate limiter with zero delay does not slow crawling
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'crawl.pages_crawled' not available on result type
}
